apps/files/lib/Command/RepairTree.php

```diff
@@ -51,9 +54,48 @@ protected function configure() {
 	}
 
 	public function execute(InputInterface $input, OutputInterface $output): int {
-		$rows = $this->findBrokenTreeBits();
 		$fix = !$input->getOption('dry-run');
 
+		$this->repairParent($fix, $output);
+		$this->repairPaths($fix, $output);
+
+		return 0;
+	}
+
+	private function repairParent(bool $fix, OutputInterface $output) {
+		$rows = $this->findMissingParents();
+
+		$output->writeln("Found " . count($rows) . " file entries with an invalid parent");
+
+		if ($fix) {
+			$this->connection->beginTransaction();
+		}
+
+		$query = $this->connection->getQueryBuilder();
+		$query->update('filecache')
+			->set('parent', $query->createParameter('parent'))
+			->where($query->expr()->eq('fileid', $query->createParameter('fileid')));
```

Comment:
```suggestion
			$query = $this->connection->getQueryBuilder();
			$query->update('filecache')
				->set('parent', $query->createParameter('parent'))
				->where($query->expr()->eq('fileid', $query->createParameter('fileid')));
		}
```
The query is only used for fixing, no?